drizzlepac/haputils/mvm_quality_analysis.py

```diff
@@ -103,49 +116,94 @@ def report_wcsname(total_product_list, json_timestamp=None, json_time_since_epoc
-            y = ypos.tolist()
+        # Loop over all the individual exposures in the list which comprise the layer
+        for edp_object in total_product.edp_list:
             
-            # Load the dictionary with the collected data for this exposure
-            active_wcs_dict = {'filename': edp_object.full_filename,
-                               'primary_wcsname': wcsname,
-                               'RA': ra,
-                               'Dec': dec,
-                               'X': x,
-                               'Y': y}
-
-            diagnostic_obj.add_data_item(active_wcs_dict, 'PrimaryWCS' + str(counter),
-                                         descriptions={'filename': 'Exposure filename',
-                                                       'primary_wcsname': 'Active WCS',
-                                                       'RA': 'Right Ascension of S_REGION Polygon ICRS which defines footprint outline',
-                                                       'Dec': 'Declination of S_REGION Polygon ICRS which defines footprint outline',
-                                                       'X': 'X position of S_REGION Polygon which defines footprint outline',
-                                                       'Y': 'Y position of S_REGION Polygon which defines footprint outline'},
-                                         units={'filename': 'unitless',
-                                                'primary_wcsname': 'unitless',
-                                                'RA': 'degrees',
-                                                'Dec': 'degrees',
-                                                'X': 'pixels',
-                                                'Y': 'pixels'})
+            # Open the exposure file
+            exp = fits.open(edp_object.full_filename)
+
+            # Each chip belonging to the same exposure/image in a filter-level product
+            # will be assigned a value of 2**n so the exposure/image can be uniquely identified
+            expo_img_value = 2**counter
+
+            # Determine the number of extensions ...
+            sci_extns = wcs_functions.get_extns(exp)
+            if len(sci_extns) == 0 and '_single' in edp_object.full_filename:
+                sci_extns = [0]
+
+            # ... and loop over the SCI extensions (each SCI extension is a chip)
+            for sci in sci_extns:
+                wcs = HSTWCS(exp, ext=sci)
+
+                # Get the WCSNAME for this specific exposure/chip
+                wcsname = exp[sci].header['WCSNAME'].upper()
+
+                # Each chip is assigned the 2**n value associated with its WCSNAME
+                suffix = wcsname.split('-')[1]
+                expo_wcs_value = wcs_pref_list[suffix]
+
+                # If this is a multi-chip exposure, get the CHIP number
+                chip_number = 0
+                try:
+                    chip_number = exp[sci].header['CCDCHIP']
```

Comment:
Just as a note regarding the impact of this on potential future development:  WFPC2 images use the 'DETECTOR' keyword in the SCI headers to record the 'chip_number'.

Reply:
Good comment!